File: go/problem_85.go
package main

func selectByBit(x, y, b int) int {
	return x*b + y*(1-b)
}

func problem85() {
	println("Problem 85:")
	assertEqual(selectByBit(10, 20, 1), 10)
	assertEqual(selectByBit(10, 20, 0), 20)
	assertEqual(selectByBit(5, 7, 1), 5)
	println("All Tests Passed")
}

File: rust/problem_85.rs
pub fn select_by_bit(x: i64, y: i64, b: i64) -> i64 { x * b + y * (1 - b) }

pub fn problem85() {
    println!("Problem 85:");
    assert_eq!(select_by_bit(10, 20, 1), 10);
    assert_eq!(select_by_bit(10, 20, 0), 20);
    assert_eq!(select_by_bit(5,   7, 1),  5);
    println!("All Tests Passed");
}
